client/sites/staging-site/components/staging-site-sync-modal/index.tsx

```diff
@@ -133,48 +145,104 @@ const getSyncConfig = ( type: 'pull' | 'push' ): SyncConfig => {
 	};
 };
 
-export default function SyncModal( { onClose, syncType, environment, siteSlug }: SyncModalProps ) {
+export default function SyncModal( {
+	onClose,
+	syncType,
+	environment,
+	siteSlug,
+	productionSiteId,
+	stagingSiteId,
+}: SyncModalProps ) {
+	const dispatch = useDispatch();
 	const syncConfig = getSyncConfig( syncType );
 
 	// TODO: Once we use the component in the Dashbaord V2, let's get siteSlug from Router instead of the passed prop
 	//const { siteSlug } = siteRoute.useParams();
 
+	const querySiteId =
+		( environment === 'staging' && syncType === 'push' ) ||
+		( environment === 'production' && syncType === 'pull' )
+			? stagingSiteId
+			: productionSiteId;
```

Comment:
If we rebase this PR after https://github.com/Automattic/wp-calypso/pull/104587 is merged, I think we could consider simplifying this logic with something like:

```suggestion
	const querySiteId = sourceEnvironment === 'staging' ? stagingSiteId : productionSiteId;
```

Reply:
Thanks, Ivan, for the suggestion! That has been added [here](https://github.com/Automattic/wp-calypso/pull/104511/commits/465dc55c72be759f6091bccb53e2c910ef775640#diff-f35b2f1fb4fd4096dc82772ce4d3aa14a569548a3f12fdf7325090d8403b2432R167)

Reply:
Thank you for the update, Roberto!